### PR TITLE
Python 3 Update

### PR DESCRIPTION
**Pull request by grimery**
- Updated for Python 3 (tested with Python 3.6.8 on Centos 8.0)
- Added requirements.txt
- Reset config.ini
- Implement blueprint indenting and sorting (prettyprinting)
- Modified CALM blueprint endpoint
- Updated README.md

### CODE REVIEW

File: calm-integrations/git-sync-for-calm-blueprints/git-sync-for-calm-blueprints.py [hunk 1]
```diff
@@ -151,22 +151,24 @@ def git_update(blueprint_names,account_data,pc_data):
...
+            print("[WARNING] Project directory {} not found in the repository.".format(bp["project"]))
             sys.exit(1)
-        print "[INFO] Fetching BP {} details.".format(bp["name"])
-        blueprint_json = get_content_calm(bp["uuid"], pc_data)
+        print("[INFO] Fetching BP {} details.".format(bp["name"]))
+        blueprint_json_from_calm = json.loads(get_content_calm(bp["uuid"], pc_data)) # convert string received from CALM into JSON object
+        blueprint_json_pp = json.dumps(blueprint_json_from_calm, indent=4, sort_keys=True) # convert JSON object into prettyprinted JSON string
+        blueprint_json = blueprint_json_pp.encode("utf-8") # # encode prettyprinted JSON string into utf-8 for base64 encode
         for file in git_file_list:
             if file["name"] == bp["name"]+".json":
```
Comment (aalekhneema) [inline calm-integrations/git-sync-for-calm-blueprints/git-sync-for-calm-blueprints.py:161]: **Failed on my local machine** 
Traceback (most recent call last):
  File "git-sync-for-calm-blueprints.py", line 229, in <module>
    git_update(blueprint_names,account_data,pc_data)
  File "git-sync-for-calm-blueprints.py", line 161, in git_update
    if file["name"] == bp["name"]+".json":
TypeError: string indices must be integers

Comment (grimery) [reply calm-integrations/git-sync-for-calm-blueprints/git-sync-for-calm-blueprints.py:161]: What version of Python? I'm using 3.6.8 and have also tested only the prettyprint lines (157-159) in the original Python2 version of the script using Python 2.7.16.

CALM 2.9.7.

Comment (aalekhneema) [reply calm-integrations/git-sync-for-calm-blueprints/git-sync-for-calm-blueprints.py:161]: **Calm : master
Python: 3.7.5**
`(uitestenv) alekh.neema@C02W91BPHV2R git-sync-for-calm-blueprints %
(uitestenv) alekh.neema@C02W91BPHV2R git-sync-for-calm-blueprints % python --version
Python 3.7.5
(uitestenv) alekh.neema@C02W91BPHV2R git-sync-for-calm-blueprints % python git-sync-for-calm-blueprints.py
/Users/alekh.neema/CALM/ENV/uitestenv/lib/python3.7/site-packages/requests/__init__.py:91: RequestsDependencyWarning: urllib3 (1.25.7) or chardet (3.0.4) doesn't match a supported version!
  RequestsDependencyWarning)
/Users/alekh.neema/CALM/ENV/uitestenv/lib/python3.7/site-packages/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning,
/Users/alekh.neema/CALM/ENV/uitestenv/lib/python3.7/site-packages/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning,
/Users/alekh.neema/CALM/ENV/uitestenv/lib/python3.7/site-packages/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning,
[INFO] Fetching BP new_single details.
/Users/alekh.neema/CALM/ENV/uitestenv/lib/python3.7/site-packages/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning,
Traceback (most recent call last):
  File "git-sync-for-calm-blueprints.py", line 229, in <module>
    git_update(blueprint_names,account_data,pc_data)
  File "git-sync-for-calm-blueprints.py", line 161, in git_update
    if file["name"] == bp["name"]+".json":
TypeError: string indices must be integers`

Comment (aalekhneema) [reply calm-integrations/git-sync-for-calm-blueprints/git-sync-for-calm-blueprints.py:161]: target repo is blank and response is `: {'message': 'This repository is empty.', 'documentation_url': 'https://developer.github.com/v3/repos/contents/#get-contents'}`. If statement to handle the same `if  "message" in git_file_list and git_file_list["message"] == "Not Found":` needs correction.

Comment (grimery) [reply calm-integrations/git-sync-for-calm-blueprints/git-sync-for-calm-blueprints.py:161]: Did you setup your github repo per the instructions in the README.md?

repo-name/<project-name-in-calm-and-config.ini>/blueprints/

Screenshot here: https://github.com/nutanix/blueprints/blob/master/calm-integrations/git-sync-for-calm-blueprints/Screenshots/github_output.png



## Pre-Requisites
* Existing github account.
	* create a repository in the github account with directory structure given below (default and project1) are projects in calm.
```.
├── default
│   └── blueprints
│       └── README.md
└── project1
    └── blueprints
        └── README.md

Comment (grimery) [reply calm-integrations/git-sync-for-calm-blueprints/git-sync-for-calm-blueprints.py:161]: Could you update requests and try again?

You have an error in your output that I'm not getting with a fresh install of Python 3.7.5 followed by requests and configparser install in requirements.txt.

I've tried replicating the error by writing to an empty repository and I am not able to replicate. I get the message that my project directory cannot be found as expected. Please let me know your results. I've now tested with Python 3.6.8, 3.7.5, and 3.8.3.

> /Users/alekh.neema/CALM/ENV/uitestenv/lib/python3.7/site-packages/requests/init.py:91: RequestsDependencyWarning: urllib3 (1.25.7) or chardet (3.0.4) doesn't match a supported version!



```
/usr/local/bin/python3.7 -V
Python 3.7.5

/usr/local/bin/python3.7 -m pip freeze
certifi==2020.4.5.1
chardet==3.0.4
configparser==5.0.0
idna==2.9
requests==2.23.0
urllib3==1.25.9
```

Comment (aalekhneema) [reply calm-integrations/git-sync-for-calm-blueprints/git-sync-for-calm-blueprints.py:161]: My bad . Sorry. Missed pre-requisites.
`if "message" in git_file_list and git_file_list["message"] == "Not Found"` could still be changed to `if "message" in git_file_list and git_file_list["message"] in ["Not Found","This repository is empty."]`  though.

**Pull request merged**